### ISSUE TITLE
Provide constructor classes for the various types?

### ISSUE THREAD

**Issue by ajvincent** [TheSpicyMeatball/jsdoc-parse-plus#1]
In ES6 modules, I came up with this, and I think it's fairly close:

[experiment.mjs.txt](https://github.com/TheSpicyMeatball/jsdoc-parse-plus/files/6289123/experiment.mjs.txt)
Unfortunately I don't know TypeScript, and I haven't checked out your repository.  So I figured I'd throw it over the wall and see if you could modify it to be more accurate and pass round-trip tests (parsing what you stringify, stringify what you parse).  Then you might be able to export these classes.

**Comment by TheSpicyMeatball** [comment]
@ajvincent I checked out your file. I'm not sure what you're trying to accomplish that you can't already do with `toCommentString` and `parse`? 

`parse` will return an object which you can put into `toCommentString` and `toCommentString` will output a string you can put into `parse`

**Comment by ajvincent** [comment]
I was trying to generate comment strings when I had started from JavaScript objects instead of - as you did - from parsing strings.  This was based on the project's page claim saying users could "programmatically generate jsdoc comments from JavaScript data":  if I provided the data, it should generally work.

As-is, if I didn't provide a `raw` property, then it blows up.  The classes are supposed to fill that gap by generating the `raw` property dynamically.

In any case, I've crafted my own solution for my project which is more tailored for my specific needs.
